### ISSUE TITLE
Change storage for webhook events

### ISSUE THREAD

**Issue by htdat** [Automattic/woocommerce-payments#5374]
From p1671520152621799-slack-C0208C3BXHP

### Description

Currently, [webhook reliably service class](https://github.com/automattic/woocommerce-payments/blob/5.1.0/includes/class-wc-payments-webhook-reliability-service.php) is using transients as a storage for webhook events.

However, with a nature of WordPress with many types of hosting, caching, plugins, transients appear not stable enough for our purpose. 

@james-allan got a specific case for this:  

> I was on a customer’s site I noticed they have the [Breeze cache plugin](https://wordpress.org/plugins/breeze/) running and from past experience with these kind of plugins, I know they can do some strange things with transients like treating the expiration as a timestamp and not a number of seconds. This particular one seems to suggest it purges expired transients but the database query to me looks like it just deletes them all.

### Acceptance criteria

- Build up a persistent storage for webhook events. 
- As mentioned in the discussion (linked above), the most suitable option is to use a custom post type as a storage. 

### Testing instructions

TBA

### Dev notes

TBA

### Additional context

TBA

**Comment by zmaglica** [comment]
Before we start to work on this issue, we need to decide what storage mechanism we should use. I guess that we can use database to store webhook events, and we can clean them afterward by using cron tasks or manually once certain event is no longer needed.

**Comment by htdat** [comment]
I think @RadoslavGeorgiev is working on this issue for the cooldown week of Gamma sprint 2
https://github.com/Automattic/woocommerce-payments/pull/5407

Moving this issue to Gamma backlog and connecting it with PR 5407.

**Comment by frosso** [comment]
Dug into whether this is still worth doing, and I don't think it is anymore - recommending we close.

A few things:
- The code never moved off transients. `WC_Payments_Webhook_Reliability_Service` still uses `set_transient`/`get_transient`, unchanged since 2023.
- The CPT attempt (#5407) was closed unmerged in Oct 2024, after sitting incomplete for ~14 months.
- The sibling issue this was meant to unblock (#5188) is now closed, and the direction there is to handle webhook delay / duplicate charges server-side rather than reworking client storage.

The fragility of transients I think is accurate, but the transient is just an ephemeral handoff between two ActionScheduler jobs that are both scheduled immediately.
So it's only at risk if a caching plugin wipes _all_ transients in _that_ short window (which is the Breeze case that kicked this off).

If we ever revive it, I _think_ the cleaner fix is passing the event data straight through ActionScheduler's `extended_args` instead of a transient + ID lookup - removes the handoff gap entirely, no new storage needed.

@htdat closing as superseded by the server-side approach, per the discussion on #5188 - shout if you'd rather keep it open.
